pkg/sidecar/util_posix.go

```diff
@@ -5,23 +5,54 @@ package sidecar
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/workloadapi"
 	"golang.org/x/sys/unix"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // RunDaemon starts the main loop
 // Starts the workload API client to listen for new SVID updates
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
-	err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
-	if err != nil && !errors.Is(err, context.Canceled) {
-		return err
+	var wg sync.WaitGroup
+
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Fatalf("Error watching X.509 context: %w", err)
+			}
+		}()
+	}
+
+	if s.config.JWKFilename != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchJWTBundles(ctx, &JWTBundlesWatcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Fatalf("Error watching JWT bundles updates: %w", err)
+			}
+		}()
 	}
 
+	if s.config.JWTFilename != "" && s.config.JWTAudience != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			s.updateJWTSVID(ctx, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
```

Comment:
Set the `unix://` prefix directly in the `config.AgentAddress` value. This will streamline the code by removing the need to prepend it in multiple places.